# piecharts for ever detected proteins from Cys and Lys experiments

# PANTHER analysis of protein class

"""
Created on 11.23.19
for Project MAP
@author: mfpfox
"""

citation
```
PANTHER version 14: more genomes, a new PANTHER GO-slim and improvements in enrichment analysis tools.
Huaiyu Mi, Anushya Muruganujan, Dustin Ebert, Xiaosong Huang and Paul D. Thomas
Nucl. Acids Res. (2019) doi: 10.1093/nar/gky1038
```

In [1]:
import os
import sys
import pandas as pd

sys.path.append("/Users/mariapalafox/Desktop/Toolbox")
from all_funx import *

from IPython.display import display, HTML
pd.set_option('display.max_columns', None)
pd.options.display.max_seq_items = 2000
display(HTML("<style>.container {width:90% !important;}</style>"))

---

# version 1 pie n=3963

**below function in toolbox "Pmap_pie.py"**

In [2]:
os.chdir("/Users/mariapalafox/Box Sync/CODE_DATA/dir_MAPpaper/CYS_LYS_2012_to2018UKBpositions/REFERENCE_2018_mappedCK/SUBSETS/")

In [3]:
# ukb file with be one annotated
everC = "CYS2018_everlabeled_gene_valueCounts_2915.csv"
everK = "LYS2018_everlabeled_gene_valueCounts_2623.csv"
uni = "countsfromfastaCCDS_filter_counts_all_AA.csv"
shared3953 = "UKBID_3953set.csv"

cys = read_pd(everC)
lys = read_pd(everK)
ukb = read_pd(uni)
dropUnnamed(ukb)

shared = read_pd(shared3953)

In [4]:
print("ALL 3963 uniprot IDs: ")
print("total of all lysines detected: ", lys.counts.sum())
print("total of all cysteine detected: ", cys.counts.sum())

ALL 3963 uniprot IDs: 
total of all lysines detected:  9042
total of all cysteine detected:  6234


In [5]:
# testing main()
Cls = cys.ID.tolist()
Kls = lys.ID.tolist()

ukbC = addcolumnconditional(Cls, ukb, 'ID', 'ever_C_ID')
ukbCK = addcolumnconditional(Kls, ukbC, 'ID', 'ever_K_ID')
print("shape ukbCK labeled = ", ukbCK.shape)

# only C detected proteins
Conly = ukbCK[(ukbCK['ever_C_ID'] == "True") & (ukbCK['ever_K_ID'] == "False")]
print("TOTAL protiens with only Cys detected: ")
print(Conly.shape)
print("TOTAL Cys in the proteins: ", Conly.C.sum())
print()

# only K detected proteins
Konly = ukbCK[(ukbCK['ever_C_ID'] == "False") & (ukbCK['ever_K_ID'] == "True")]
print("TOTAL protiens with only Lysine detected: ")
print(Konly.shape)
print("TOTAL Lysine in the proteins: ", Konly.K.sum())
print()

# both C K detecte proteins
bothck = ukbCK[(ukbCK['ever_C_ID'] == "True") & (ukbCK['ever_K_ID'] == "True")]
print("TOTAL protiens with both CK detected: ")
print(bothck.shape)
print("TOTAL Lysine in the proteins: ", bothck.K.sum())
print("TOTAL Cysteine in the proteins: ", bothck.C.sum())
print()

# all C IDs
Call = ukbCK[ukbCK['ever_C_ID'] == "True"]
print("TOTAL protiens with all C detected: ")
print(Call.shape)
print("TOTAL C in the proteins: ", Call.C.sum())
print()

Kall = ukbCK[ukbCK['ever_K_ID'] == "True"]
print("TOTAL protiens with all K detected: ")
print(Kall.shape)
print("TOTAL K in the proteins: ", Kall.K.sum())


checking labeled column values: 
  ever_C_ID  Count
0     False  15517
1      True   2915

checking labeled column values: 
  ever_K_ID  Count
0     False  15809
1      True   2623

shape ukbCK labeled =  (18432, 30)
TOTAL protiens with only Cys detected: 
(1340, 30)
TOTAL Cys in the proteins:  19812

TOTAL protiens with only Lysine detected: 
(1048, 30)
TOTAL Lysine in the proteins:  34910

TOTAL protiens with both CK detected: 
(1575, 30)
TOTAL Lysine in the proteins:  61917
TOTAL Cysteine in the proteins:  14944

TOTAL protiens with all C detected: 
(2915, 30)
TOTAL C in the proteins:  34756

TOTAL protiens with all K detected: 
(2623, 30)
TOTAL K in the proteins:  96827


---
---
---


In [7]:
shared.head(3)

ID
0  A0AVT1
1  A0FGR8
2  A0JNW5

In [8]:
cys.head(3)

ID  counts  final_refID
0  P49327      25         True
1  P21333      25         True
2  Q15149      22         True

In [9]:
lys.head(3)

ID  counts  final_refID
0  Q09666     105         True
1  P21333      50         True
2  O75369      47         True

# version of PIE chart with 3,953 IDs shared across all ensembl releases

In [10]:
# 3953 ID version 
fil = shared.ID.tolist()
print("length of list: ", len(fil))
print("length of set: ", len(set(fil)))
print()
lysdrop = addcolumnconditionalDrop(fil, lys, 'ID', 'set3953_ID')
cysdrop = addcolumnconditionalDrop(fil, cys, 'ID', 'set3953_ID')
print("shape lys drop  = ", lysdrop.shape)
print("shape cys drop  = ", cysdrop.shape)

length of list:  3953
length of set:  3953

  set3953_ID  Count
0       True   2618
1      False      5
dropping false
df shape post drop:  (2618, 4)
  set3953_ID  Count
0       True   2909
1      False      6
dropping false
df shape post drop:  (2909, 4)
shape lys drop  =  (2618, 4)
shape cys drop  =  (2909, 4)


In [13]:
idsimple = pd.DataFrame(fil)
idsimple.columns = ['ID']
idsimple.shape

(3953, 1)

In [14]:
# saving for later
idsimple.to_csv("/Users/mariapalafox/Box Sync/CODE_DATA/dir_MAPpaper/id3953.csv", index=False)

In [53]:
print("ALL 3953 uniprot IDs: ")
print("total of all lysines detected: ", lysdrop.counts.sum())
print("total of all cysteines detected: ", cysdrop.counts.sum())

ALL 3953 uniprot IDs: 
total of all lysines detected:  9034
total of all cysteines detected:  6225


In [51]:
# change df name
Cls = cysdrop.ID.tolist()
Kls = lysdrop.ID.tolist()

ukbC = addcolumnconditional(Cls, ukb, 'ID', 'ever_C_ID')
ukbCK = addcolumnconditional(Kls, ukbC, 'ID', 'ever_K_ID')
print("shape ukbCK labeled = ", ukbCK.shape)

# only C detected proteins
Conly = ukbCK[(ukbCK['ever_C_ID'] == "True") & (ukbCK['ever_K_ID'] == "False")]
print("TOTAL protiens with only Cys detected: ")
print(Conly.shape)
print("TOTAL Cys in the proteins: ", Conly.C.sum())
print()

# only K detected proteins
Konly = ukbCK[(ukbCK['ever_C_ID'] == "False") & (ukbCK['ever_K_ID'] == "True")]
print("TOTAL protiens with only Lysine detected: ")
print(Konly.shape)
print("TOTAL Lysine in the proteins: ", Konly.K.sum())
print()

# both C K detecte proteins
bothck = ukbCK[(ukbCK['ever_C_ID'] == "True") & (ukbCK['ever_K_ID'] == "True")]
print("TOTAL protiens with both CK detected: ")
print(bothck.shape)
print("TOTAL Lysine in the proteins: ", bothck.K.sum())
print("TOTAL Cysteine in the proteins: ", bothck.C.sum())
print()

# all C IDs
Call = ukbCK[ukbCK['ever_C_ID'] == "True"]
print("TOTAL protiens with all C detected: ")
print(Call.shape)
print("TOTAL C in the proteins: ", Call.C.sum())
print()

Kall = ukbCK[ukbCK['ever_K_ID'] == "True"]
print("TOTAL protiens with all K detected: ")
print(Kall.shape)
print("TOTAL K in the proteins: ", Kall.K.sum())


checking labeled column values: 
  ever_C_ID  Count
0     False  15523
1      True   2909

checking labeled column values: 
  ever_K_ID  Count
0     False  15814
1      True   2618

shape ukbCK labeled =  (18432, 30)
TOTAL protiens with only Cys detected: 
(1335, 30)
TOTAL Cys in the proteins:  19768

TOTAL protiens with only Lysine detected: 
(1044, 30)
TOTAL Lysine in the proteins:  34814

TOTAL protiens with both CK detected: 
(1574, 30)
TOTAL Lysine in the proteins:  61905
TOTAL Cysteine in the proteins:  14940

TOTAL protiens with all C detected: 
(2909, 30)
TOTAL C in the proteins:  34708

TOTAL protiens with all K detected: 
(2618, 30)
TOTAL K in the proteins:  96719


---
---
---


---
---
---


# PANTHER PROTEIN CLASSIFICATION

version = PANTHER14.1 is generated from the **2018_04 release of ReferenceProteome dataset**

<http://www.pantherdb.org/chart/pantherChart.jsp?listType=1&filterLevel=1&type=5&chartType=2&save=yes&basketItems=all&zoom=1&trackingId=626614EA3D2EAB0581A6BA3EDA4F01F5>

### unable to map some IDs
#### from cys ever n = 2915
```
 Unable to map following ids:
# UNIQUE TO C EVER LIST
Q14116

# IN LYS MISMAP LIST (5)
P50135
Q15121
Q9Y3B4
P42167
P61513
```

#### from lys ever n = 2623
```
 Unable to map following ids:
# UNIQUE TO K EVER LIST
Q13765
Q99470
A6ZKI3
Q9GZP8
P60896
P42773
Q9NRS6
O60656

# IN CYS MISMAP LIST (5)
P61513
P42167
Q9Y3B4
Q15121
P50135
```

| Not mapped in C or K ever ID list | protein primary name from ukb web |
|------------------------------------|-------------------------------------|
| P61513 | 60S ribosomal protein L37a, gene(RPL37A) |
| P42167 | Lamina-associated polypeptide 2, isoforms beta/gamma, gene(TMPO) |
| Q9Y3B4 | Splicing factor 3B subunit 6, gene(SF3B6) |
| Q15121 | Astrocytic phosphoprotein PEA-15, gene(PEA15) |
| P50135 | Histamine N-methyltransferase, gene(HNMT) |


## Ontologies PANTHER™ GO slim (version 14.1, based on GO release 2018-07-03, released 2019-03-12)
- 3039 total terms
- 2004 biological process terms
- 512 cellular component terms
- 523 molecular function terms
- PANTHER™ Protein Class (version 14.1, released 2019-03-12)
    - 256 total terms
    
    
Accessed : PANTHER™ Data at a glance on 12.15.19

## Unable to map following ids (14 not found out of 3953):
```
A6ZKI3
O60656
P42167
P42773
P50135
P60896
P61513
Q13765
Q14116
Q15121
Q99470
Q9GZP8
Q9NRS6
Q9Y3B4
```


In [54]:
os.chdir("/Users/mariapalafox/Desktop/myMAP_notes/PANTHER/v14PANTHER/")
print(os.listdir())

['id3953.csv', 'PANTHERv14_ProteinRelationship_MAJORclasses.txt', '.DS_Store', 'PANTHERv14_3939_of_3953_UKBIDS_EverlabeledCK.txt', 'miscast_24Protein_classes.csv', 'pantherChart_3939IDs_barchart.txt', 'PANTHERv14_3939_simpledf.csv', 'pantherChart_3939_barchart_protein_class.txt', 'PANTHERv14_3953_14notFount_total3939.txt', 'PANTHERv14_Protein_class_ontology_relatioship.txt', 'PANTHERv14_3939_of_3953_UKBIDS_EverlabeledCK.csv', 'panther_protein_class_value_counts_388.csv', 'biomart_v98_genedescription_21355.txt', 'PTHR14.1_human.txt']


In [56]:
panther3953 = pd.read_csv("PANTHERv14_MERGEME_3953.csv")



In [85]:
panther3953



Mapped_IDs                                    GeneName_Symbol  \
0        O14737     Programmed cell death protein 5;PDCD5;ortholog   
1        P62195  26S proteasome regulatory subunit 8;PSMC5;orth...   
2        Q8IYB3  Serine/arginine repetitive matrix protein 1;SR...   
3        Q9UMY4                    Sorting nexin-12;SNX12;ortholog   
4        Q12849            G-rich sequence factor 1;GRSF1;ortholog   
...         ...                                                ...   
3948     Q15121                                                NaN   
3949     Q99470                                                NaN   
3950     Q9GZP8                                                NaN   
3951     Q9NRS6                                                NaN   
3952     Q9Y3B4                                                NaN   

                                  PANTHER_fam_subfamily  \
0       PROGRAMMED CELL DEATH PROTEIN 5 (PTHR10840:SF1)   
1     26S PROTEASOME REGULATORY SUBUNIT 8 (PTHR23073...   
2     SERINE/ARGININE REPETITIVE MATRIX PROTEIN 1 (P...   
3                      SORTING NEXIN-12 (PTHR45963:SF3)   
4             G-RICH SEQUENCE FACTOR 1 (PTHR13976:SF42)   
...                                                 ...   
3948                                                NaN   
3949                                                NaN   
3950                                                NaN   
3951                                                NaN   
3952                                                NaN   

                                     Molecular_function  \
0                                                   NaN   
1                 TBP-class protein binding(GO:0017025)   
2                              mRNA binding(GO:0003729)   
3                                                   NaN   
4     nucleic acid binding(GO:0003676);structural co...   
...                                                 ...   
3948                                                NaN   
3949                                                NaN   
3950                                                NaN   
3951                                                NaN   
3952                                                NaN   

                                     Biological_process  \
0                                                   NaN   
1     RNA polymerase II transcriptional preinitiatio...   
2                           mRNA processing(GO:0006397)   
3                                                   NaN   
4            mRNA splicing, via spliceosome(GO:0000398)   
...                                                 ...   
3948                                                NaN   
3949                                                NaN   
3950                                                NaN   
3951                                                NaN   
3952                                                NaN   

                                     Cellular_component  \
0               cytosol(GO:0005829);nucleus(GO:0005634)   
1     cytosolic part(GO:0044445);nuclear lumen(GO:00...   
2                                                   NaN   
3                                                   NaN   
4                                                   NaN   
...                                                 ...   
3948                                                NaN   
3949                                                NaN   
3950                                                NaN   
3951                                                NaN   
3952                                                NaN   

                        Protein_class  \
0           enzyme modulator(PC00095)   
1                  hydrolase(PC00121)   
2     mRNA processing factor(PC00147)   
3                                 NaN   
4          ribosomal protein(PC00202)   
...                               ...   
3948                              NaN   


In [58]:
describeMe(panther3953)

(3953, 15)
Index(['Mapped_IDs', 'GeneName_Symbol', 'PANTHER_fam_subfamily',
       'Molecular_function', 'Biological_process', 'Cellular_component',
       'Protein_class', 'Pathway', 'public_start', 'public_end', 'chr',
       'GOdb_MF_complete', 'GOdb_BP_complete', 'GOdb_CC_complete',
       'Reactome_pathway'],
      dtype='object')
Mapped_IDs                  0
GeneName_Symbol            14
PANTHER_fam_subfamily      14
Molecular_function       1666
Biological_process       1590
Cellular_component       1821
Protein_class            1964
Pathway                  3227
public_start               23
public_end                 23
chr                        23
GOdb_MF_complete          134
GOdb_BP_complete          172
GOdb_CC_complete          122
Reactome_pathway         1222
dtype: int64


In [59]:
#panther3939value = panther3939['Protein_class'].value_counts()
#panther3939value.to_csv("panther_protein_class_value_counts_388.csv")

In [60]:
3953-1964

1989

# 1964 out of 3953 IDs dont have a specific Protein class from PANTHERv14

## [1] assign 1989 with Protein class from PANTHER to 1 of 24 MAJOR protein classes
## [2] other 1964 IDs with no protein class information, merge with ENSMEBL file for more details
## [3] assign merge file to 1 of 24 MAJOR protein classes
- miscast protein class rule:
1. Ensembl family description
2. molecular funcx/biological process annotation previous collected from PANTHER or 
3. molecular funcx/biological process annotation available in UniProt as definded by teh gene ontology consortium

In [25]:
# miscast 24 protein classes for 1330 genes
miscast = pd.read_csv("/Users/mariapalafox/Desktop/myMAP_notes/MISCAST_BROAD/miscast_24Protein_classes.csv")
miscast

Protein Class  Gene count  Structure count  \
0                   All proteins        1330            14270   
1        Calcium-Binding Protein          35              311   
2         Cell Adhesion Molecule          40              298   
3          Cell Junction Protein          24              128   
4                      Chaperone          27              137   
5           Cytoskeletal Protein          83              635   
6       Defense/Immunity protein          57             1069   
7               Enzyme Modulator         119             1280   
8   Extracellular Matrix Protein          41              309   
9                      Hydrolase         198             2674   
10                     Isomerase          20              181   
11                        Kinase         126             1994   
12                        Ligase          54              347   
13                         Lyase          40              993   
14      Membrane Traffic Protein          21               96   
15          Nucleic Acid Binding         258             2557   
16                Oxidoreductase         131              958   
17                   Phosphatase          21              144   
18                      Protease          78             1491   
19                      Receptor         178             2186   
20            Signaling Molecule         134             1236   
21          Transcription Factor         124             1521   
22      Transfer/Carrier Protein          31              842   
23                   Transferase         212             2509   
24                   Transporter          85              862   

    Population variants mapped on the structure  \
0                                        164915   
1                                          3834   
2                                          5494   
3                                          2395   
4                                          3187   
5                                         10612   
6                                          7829   
7                                         14839   
8                                          4883   
9                                         33556   
10                                         2988   
11                                        17994   
12                                         8024   
13                                         6258   
14                                         2352   
15                                        28496   
16                                        18016   
17                                         2773   
18                                        12485   
19                                        23385   
20                                        13438   
21                                         7527   
22                                         3529   
23                                        32215   
24                                        11939   

    Pathogenic variants mapped on the structure  \
0                                         32923   
1                                          1007   
2                                          2523   
3                                           472   
4                                           231   
5                                          2219   
6                                          1442   
7                                          2418   
8                                          1380   
9                                          8116   
10                                          694   
11                                         3768   
12                                          986   
13                                          885   
14                                          126   
15                                         4378   
16                                         3140   
17                                          395   
18                              

In [65]:
barsets = pd.read_csv("/Users/mariapalafox/Desktop/myMAP_notes/PANTHER/v14PANTHER/CLASS_subsets_PANTHERv14/PC_allclasses.txt", 
                      delimiter='\t', header=None)
barsets



0       1  \
0     PC00060|HGNC=26032|UniProtKB=Q7L5N7  Q7L5N7   
1      PC00060|HGNC=9064|UniProtKB=Q9UPR0  Q9UPR0   
2     PC00060|HGNC=17847|UniProtKB=Q15208  Q15208   
3       PC00060|HGNC=392|UniProtKB=P31751  P31751   
4     PC00060|HGNC=15917|UniProtKB=Q9NQ66  Q9NQ66   
...                                   ...     ...   
2658  PC00227|HGNC=16695|UniProtKB=P51572  P51572   
2659  PC00227|HGNC=14579|UniProtKB=Q9NRW7  Q9NRW7   
2660   PC00227|HGNC=9002|UniProtKB=P48739  P48739   
2661  PC00227|HGNC=24131|UniProtKB=Q9UHQ4  Q9UHQ4   
2662    PC00227|HGNC=856|UniProtKB=P21283  P21283   

                                                      2  \
0     Lysophosphatidylcholine acyltransferase 2;LPCA...   
1     Inactive phospholipase C-like protein 2;PLCL2;...   
2     Serine/threonine-protein kinase 38;STK38;ortholog   
3     RAC-beta serine/threonine-protein kinase;AKT2;...   
4     1-phosphatidylinositol 4,5-bisphosphate phosph...   
...                                                 ...   
2658  B-cell receptor-associated protein 31;BCAP31;o...   
2659  Vacuolar protein sorting-associated protein 45...   
2660  Phosphatidylinositol transfer protein beta iso...   
2661  B-cell receptor-associated protein 29;BCAP29;o...   
2662  V-type proton ATPase subunit C 1;ATP6V1C1;orth...   

                                                      3  \
0     LYSOPHOSPHATIDYLCHOLINE ACYLTRANSFERASE 2 (PTH...   
1     INACTIVE PHOSPHOLIPASE C-LIKE PROTEIN 2 (PTHR1...   
2     SERINE/THREONINE-PROTEIN KINASE 38 (PTHR24356:...   
3     RAC-BETA SERINE/THREONINE-PROTEIN KINASE (PTHR...   
4     1-PHOSPHATIDYLINOSITOL 4,5-BISPHOSPHATE PHOSPH...   
...                                                 ...   
2658  B-CELL RECEPTOR-ASSOCIATED PROTEIN 31 (PTHR127...   
2659  VACUOLAR PROTEIN SORTING-ASSOCIATED PROTEIN 45...   
2660  PHOSPHATIDYLINOSITOL TRANSFER PROTEIN BETA ISO...   
2661  B-CELL RECEPTOR-ASSOCIATED PROTEIN 29 (PTHR127...   
2662   V-TYPE PROTON ATPASE SUBUNIT C 1 (PTHR10137:SF5)   

                                                      4             5  
0     acyltransferase(PC00042);annexin(PC00050);calm...  Homo sapiens  
1     calcium-binding protein(PC00060);guanyl-nucleo...  Homo sapiens  
2     annexin(PC00050);calmodulin(PC00061);non-recep...  Homo sapiens  
3     annexin(PC00050);calmodulin(PC00061);non-recep...  Homo sapiens  
4     calcium-binding protein(PC00060);guanyl-nucleo...  Homo sapiens  
...                                                 ...           ...  
2658  membrane traffic protein(PC00150);transporter(...  Homo sapiens  
2659  membrane trafficking regulatory protein(PC0015...  Homo sapiens  
2660  transfer/carrier protein(PC00219);transporter(...  Homo sapiens  
2661  membrane traffic protein(PC00150);transporter(...  Homo sapiens  
2662           ATP synthase(PC00002);hydrolase(PC00121)  Homo sapiens  

[2663 rows x 6 columns]

In [66]:
bset = barsets[[0,1]].copy()
bset



0       1
0     PC00060|HGNC=26032|UniProtKB=Q7L5N7  Q7L5N7
1      PC00060|HGNC=9064|UniProtKB=Q9UPR0  Q9UPR0
2     PC00060|HGNC=17847|UniProtKB=Q15208  Q15208
3       PC00060|HGNC=392|UniProtKB=P31751  P31751
4     PC00060|HGNC=15917|UniProtKB=Q9NQ66  Q9NQ66
...                                   ...     ...
2658  PC00227|HGNC=16695|UniProtKB=P51572  P51572
2659  PC00227|HGNC=14579|UniProtKB=Q9NRW7  Q9NRW7
2660   PC00227|HGNC=9002|UniProtKB=P48739  P48739
2661  PC00227|HGNC=24131|UniProtKB=Q9UHQ4  Q9UHQ4
2662    PC00227|HGNC=856|UniProtKB=P21283  P21283

[2663 rows x 2 columns]

In [67]:
# split column create new col 
splitdf = bset[0].str.split("|", n=2, expand=True)
splitdf.columns = ["PC_major", "HGNC", "UKB"]
splitdf
PCdf = pd.concat([bset, splitdf], axis=1)


In [69]:

PCdf = PCdf[[1, "PC_major", "HGNC"]].copy()
PCdf.head(3)


1 PC_major        HGNC
0  Q7L5N7  PC00060  HGNC=26032
1  Q9UPR0  PC00060   HGNC=9064
2  Q15208  PC00060  HGNC=17847

In [73]:
#PCdf.columns = ["UKBID", "PC_major", "HGNC"]
len(PCdf['UKBID'].unique())

def setme(x):
    return list(set(x))

PCukb = PCdf.groupby('UKBID')['PC_major'].apply(lambda x: setme(x))
PCukb


UKBID
A0AVT1             [PC00219, PC00142]
A1L0T0    [PC00176, PC00144, PC00220]
A2A288                      [PC00171]
A3KN83                      [PC00171]
A5YKK6                      [PC00218]
                     ...             
Q9Y6J9                      [PC00218]
Q9Y6M9                      [PC00176]
Q9Y6W5                      [PC00085]
Q9Y6X8             [PC00171, PC00218]
Q9Y6Y8             [PC00150, PC00121]
Name: PC_major, Length: 1989, dtype: object

In [74]:
groupUKB = pd.DataFrame(PCukb)
groupUKB

PC_major
UKBID                              
A0AVT1           [PC00219, PC00142]
A1L0T0  [PC00176, PC00144, PC00220]
A2A288                    [PC00171]
A3KN83                    [PC00171]
A5YKK6                    [PC00218]
...                             ...
Q9Y6J9                    [PC00218]
Q9Y6M9                    [PC00176]
Q9Y6W5                    [PC00085]
Q9Y6X8           [PC00171, PC00218]
Q9Y6Y8           [PC00150, PC00121]

[1989 rows x 1 columns]

In [75]:
groupUKB.to_csv("PC_major_UKBID_dict1989.csv")

In [28]:
proclasspanther = pd.read_csv("PANTHERv14_Protein_class_ontology_relatioship.txt", delimiter = '\t')
proclasspanther

PC00000 1.00.00.00.00   protein class   
0    PC00197 1.01.00.00.00   receptor    A molecula...
1    PC00021 1.01.01.00.00   G-protein coupled rece...
2    PC00194 1.01.02.00.00   protein kinase recepto...
3    PC00233 1.01.02.01.00   tyrosine protein kinas...
4    PC00205 1.01.02.02.00   serine/threonine prote...
..                                                 ...
296  PC00226 1.29.00.00.00   transmembrane receptor...
297  PC00211 1.30.00.00.00   structural protein  Ex...
298  PC00161 1.30.01.00.00   myelin protein  Protei...
299  PC00210 1.31.00.00.00   storage protein A prot...
300  PC00212 1.32.00.00.00   surfactant  A surface ...

[301 rows x 1 columns]

# importing biomart v98 ensembl gene description file download

- 12.15.19
- for labeling protein class of genes that failed in PANTHER search

## come back for PDB-ENSP mapping aa positions!

In [ ]:
biomart = pd.read_csv("biomart_v98_genedescription_21355.txt", delimiter='\t')



In [33]:
biomart.columns
biomart.columns = ['ENSG', 'ENSGv', 'Ensembl_Gene_description', 'Ensembl_Strand',
       'Ensembl_Family_Description', 'Ensembl_Protein_FamilyIDs', 'ENSP',
       'UKBID', 'PDB-ENSPmappings', 'PDB-ENSP_start', 'PDB-ENSP_end',
       'HGNC_symbol']



Index(['ENSG', 'ENSGv', 'Ensembl_Gene_description', 'Ensembl_Strand',
       'Ensembl_Family_Description', 'Ensembl_Protein_FamilyIDs', 'ENSP',
       'UniProtKB/SP_ID', 'PDB-ENSPmappings', 'PDB-ENSP_start', 'PDB-ENSP_end',
       'HGNC_symbol'],
      dtype='object')

In [38]:
biomart.head(5)



ENSG              ENSGv  \
0  ENSG00000198888  ENSG00000198888.2   
1  ENSG00000198888  ENSG00000198888.2   
2  ENSG00000198763  ENSG00000198763.3   
3  ENSG00000198763  ENSG00000198763.3   
4  ENSG00000198804  ENSG00000198804.2   

                            Ensembl_Gene_description  Ensembl_Strand  \
0  mitochondrially encoded NADH:ubiquinone oxidor...               1   
1  mitochondrially encoded NADH:ubiquinone oxidor...               1   
2  mitochondrially encoded NADH:ubiquinone oxidor...               1   
3  mitochondrially encoded NADH:ubiquinone oxidor...               1   
4  mitochondrially encoded cytochrome c oxidase I...               1   

                          Ensembl_Family_Description  \
0  NADH UBIQUINONE OXIDOREDUCTASE CHAIN 1 EC_7.1....   
1  NADH UBIQUINONE OXIDOREDUCTASE CHAIN 1 EC_7.1....   
2  NADH UBIQUINONE OXIDOREDUCTASE CHAIN 2 EC_7.1....   
3  NADH UBIQUINONE OXIDOREDUCTASE CHAIN 2 EC_7.1....   
4  CYTOCHROME C OXIDASE SUBUNIT 1 EC_1.9.3.1 CYTO...   

  Ensembl_Protein_FamilyIDs             ENSP   UKBID PDB-ENSPmappings  \
0             PTHR11432_SF3  ENSP00000354687  P03886           5xtc.s   
1             PTHR11432_SF3  ENSP00000354687  P03886           5xtd.s   
2            PTHR22773_SF41  ENSP00000355046  P03891           5xtc.i   
3            PTHR22773_SF41  ENSP00000355046  P03891           5xtd.i   
4                 PTHR10422  ENSP00000354499  P00395           5z62.A   

   PDB-ENSP_start  PDB-ENSP_end HGNC_symbol  
0             1.0         318.0      MT-ND1  
1             1.0         318.0      MT-ND1  
2             1.0         347.0      MT-ND2  
3             1.0         347.0      MT-ND2  
4             1.0         513.0      MT-CO1

In [39]:
nonredun_biomart = biomart[['ENSG', 'ENSGv', 'Ensembl_Gene_description', 'Ensembl_Strand',
       'Ensembl_Family_Description', 'Ensembl_Protein_FamilyIDs', 'ENSP',
       'UKBID', 'HGNC_symbol']].copy()

nonredun_biomart.shape

(214277, 9)

In [40]:
# drop duplicates
nonredun_biomart.drop_duplicates(inplace=True)
nonredun_biomart.shape

(47336, 9)

In [42]:
# filter biomart file for only 3953 IDs
id3953 = pd.read_csv("id3953.csv")
ls3953 = id3953.ID.tolist()
print(len(ls3953))

3953


In [43]:
nonredun_biomart.head(3)

ENSG              ENSGv  \
0  ENSG00000198888  ENSG00000198888.2   
2  ENSG00000198763  ENSG00000198763.3   
4  ENSG00000198804  ENSG00000198804.2   

                            Ensembl_Gene_description  Ensembl_Strand  \
0  mitochondrially encoded NADH:ubiquinone oxidor...               1   
2  mitochondrially encoded NADH:ubiquinone oxidor...               1   
4  mitochondrially encoded cytochrome c oxidase I...               1   

                          Ensembl_Family_Description  \
0  NADH UBIQUINONE OXIDOREDUCTASE CHAIN 1 EC_7.1....   
2  NADH UBIQUINONE OXIDOREDUCTASE CHAIN 2 EC_7.1....   
4  CYTOCHROME C OXIDASE SUBUNIT 1 EC_1.9.3.1 CYTO...   

  Ensembl_Protein_FamilyIDs             ENSP   UKBID HGNC_symbol  
0             PTHR11432_SF3  ENSP00000354687  P03886      MT-ND1  
2            PTHR22773_SF41  ENSP00000355046  P03891      MT-ND2  
4                 PTHR10422  ENSP00000354499  P00395      MT-CO1

In [44]:
biomart2 = addcolumnconditionalDrop(ls3953, nonredun_biomart, 'UKBID', 'in3953')
biomart2.shape

  in3953  Count
0  False  36823
1   True  10513
dropping false
df shape post drop:  (10513, 10)


(10513, 10)

In [45]:
biomart2

ENSG              ENSGv  \
0      ENSG00000280706  ENSG00000280706.3   
1      ENSG00000280706  ENSG00000280706.3   
2      ENSG00000280951  ENSG00000280951.3   
3      ENSG00000280951  ENSG00000280951.3   
4      ENSG00000280951  ENSG00000280951.3   
...                ...                ...   
10508  ENSG00000284944  ENSG00000284944.2   
10509  ENSG00000285023  ENSG00000285023.2   
10510  ENSG00000285023  ENSG00000285023.2   
10511  ENSG00000284901  ENSG00000284901.2   
10512  ENSG00000284869  ENSG00000284869.2   

                                Ensembl_Gene_description  Ensembl_Strand  \
0      REX4 homolog, 3'-5' exonuclease [Source:HGNC S...              -1   
1      REX4 homolog, 3'-5' exonuclease [Source:HGNC S...              -1   
2          surfeit 4 [Source:HGNC Symbol;Acc:HGNC:11476]              -1   
3          surfeit 4 [Source:HGNC Symbol;Acc:HGNC:11476]              -1   
4          surfeit 4 [Source:HGNC Symbol;Acc:HGNC:11476]              -1   
...                                                  ...             ...   
10508  mediator complex subunit 18 [Source:HGNC Symbo...               1   
10509  cysteine and histidine rich domain containing ...              -1   
10510  cysteine and histidine rich domain containing ...              -1   
10511  RuvB like AAA ATPase 1 [Source:HGNC Symbol;Acc...              -1   
10512  eukaryotic elongation factor, selenocysteine-t...               1   

                              Ensembl_Family_Description  \
0                           RNA EXONUCLEASE 4 EC_3.1.-.-   
1                           RNA EXONUCLEASE 4 EC_3.1.-.-   
2                                        SURFEIT LOCUS 4   
3                                        SURFEIT LOCUS 4   
4                                        SURFEIT LOCUS 4   
...                                                  ...   
10508  MEDIATOR OF RNA POLYMERASE II TRANSCRIPTION SU...   
10509  CYSTEINE AND HISTIDINE RICH DOMAIN CONTAINING ...   
10510  CYSTEINE AND HISTIDINE RICH DOMAIN CONTAINING ...   
10511                                  RUVB  EC_3.6.4.12   
10512  SELENOCYSTEINE SPECIFIC ELONGATION FACTOR ELON...   

      Ensembl_Protein_FamilyIDs             ENSP   UKBID HGNC_symbol in3953  
0                PTHR12801_SF49  ENSP00000487504  Q9GZR2       REXO4   True  
1                PTHR12801_SF49  ENSP00000486065  Q9GZR2       REXO4   True  
2                 PTHR23427_SF1  ENSP00000486869  O15260       SURF4   True  
3                 PTHR23427_SF1  ENSP00000486049  O15260       SURF4   True  
4                 PTHR23427_SF1  ENSP00000487537  O15260       SURF4   True  
...                         ...              ...     ...         ...    ...  
10508                 PTHR13321  ENSP00000494048  Q9BUE0       MED18   True  
10509             PTHR12621_SF9  ENSP00000496744  Q9UHD1     CHORDC1   True  
10510             PTHR12621_SF9  ENSP00000494044  Q9UHD1     CHORDC1   True  
10511                 PTHR11093  ENSP00000494621  Q9Y265      RUVBL1   True  
10512           PTHR23115_SF106  ENSP00000494933  P57772      EEFSEC   True  

[10513 rows x 10 columns]

In [47]:
len(biomart2.UKBID.unique())



3952

In [80]:
biomart3 = biomart2[['Ensembl_Gene_description', 'Ensembl_Strand',
       'UKBID', 'HGNC_symbol']].copy()
biomart3.drop_duplicates(inplace=True)
biomart3.shape

(3991, 4)

In [81]:
biomart3



Ensembl_Gene_description  Ensembl_Strand  \
0      REX4 homolog, 3'-5' exonuclease [Source:HGNC S...              -1   
2          surfeit 4 [Source:HGNC Symbol;Acc:HGNC:11476]              -1   
5      tRNA splicing endonuclease subunit 34 [Source:...               1   
7      ribosomal protein L7a [Source:HGNC Symbol;Acc:...               1   
8      tight junction protein 1 [Source:HGNC Symbol;A...              -1   
...                                                  ...             ...   
10363  C2 calcium dependent domain containing 5 [Sour...              -1   
10369  cytidine monophosphate N-acetylneuraminic acid...               1   
10371  histone cluster 3 H3 [Source:HGNC Symbol;Acc:H...              -1   
10419  RNA polymerase III subunit E [Source:HGNC Symb...              -1   
10427  methionine sulfoxide reductase A [Source:HGNC ...              -1   

        UKBID HGNC_symbol  
0      Q9GZR2       REXO4  
2      O15260       SURF4  
5      Q9BSV6      TSEN34  
7      P62424       RPL7A  
8      Q07157        TJP1  
...       ...         ...  
10363  Q86YS7       C2CD5  
10369  Q8NFW8        CMAS  
10371  Q16695     HIST3H3  
10419  Q9NVU0      POLR3E  
10427  Q9UJ68        MSRA  

[3991 rows x 4 columns]

In [88]:
panther3953.columns = ['UKBID', 'GeneName_Symbol', 'PANTHER_fam_subfamily',
       'Molecular_function', 'Biological_process', 'Cellular_component',
       'Protein_class', 'Pathway', 'public_start', 'public_end', 'chr',
       'GOdb_MF_complete', 'GOdb_BP_complete', 'GOdb_CC_complete',
       'Reactome_pathway']
panther3953

UKBID                                    GeneName_Symbol  \
0     O14737     Programmed cell death protein 5;PDCD5;ortholog   
1     P62195  26S proteasome regulatory subunit 8;PSMC5;orth...   
2     Q8IYB3  Serine/arginine repetitive matrix protein 1;SR...   
3     Q9UMY4                    Sorting nexin-12;SNX12;ortholog   
4     Q12849            G-rich sequence factor 1;GRSF1;ortholog   
...      ...                                                ...   
3948  Q15121                                                NaN   
3949  Q99470                                                NaN   
3950  Q9GZP8                                                NaN   
3951  Q9NRS6                                                NaN   
3952  Q9Y3B4                                                NaN   

                                  PANTHER_fam_subfamily  \
0       PROGRAMMED CELL DEATH PROTEIN 5 (PTHR10840:SF1)   
1     26S PROTEASOME REGULATORY SUBUNIT 8 (PTHR23073...   
2     SERINE/ARGININE REPETITIVE MATRIX PROTEIN 1 (P...   
3                      SORTING NEXIN-12 (PTHR45963:SF3)   
4             G-RICH SEQUENCE FACTOR 1 (PTHR13976:SF42)   
...                                                 ...   
3948                                                NaN   
3949                                                NaN   
3950                                                NaN   
3951                                                NaN   
3952                                                NaN   

                                     Molecular_function  \
0                                                   NaN   
1                 TBP-class protein binding(GO:0017025)   
2                              mRNA binding(GO:0003729)   
3                                                   NaN   
4     nucleic acid binding(GO:0003676);structural co...   
...                                                 ...   
3948                                                NaN   
3949                                                NaN   
3950                                                NaN   
3951                                                NaN   
3952                                                NaN   

                                     Biological_process  \
0                                                   NaN   
1     RNA polymerase II transcriptional preinitiatio...   
2                           mRNA processing(GO:0006397)   
3                                                   NaN   
4            mRNA splicing, via spliceosome(GO:0000398)   
...                                                 ...   
3948                                                NaN   
3949                                                NaN   
3950                                                NaN   
3951                                                NaN   
3952                                                NaN   

                                     Cellular_component  \
0               cytosol(GO:0005829);nucleus(GO:0005634)   
1     cytosolic part(GO:0044445);nuclear lumen(GO:00...   
2                                                   NaN   
3                                                   NaN   
4                                                   NaN   
...                                                 ...   
3948                                                NaN   
3949                                                NaN   
3950                                                NaN   
3951                                                NaN   
3952                                                NaN   

                        Protein_class  \
0           enzyme modulator(PC00095)   
1                  hydrolase(PC00121)   
2     mRNA processing factor(PC00147)   
3                                 NaN   
4          ribosomal protein(PC00202)   
...                               ...   
3948                              NaN   
3949                              NaN 

In [82]:
len(biomart3['UKBID'].unique())



3952

In [83]:
checkColumnValues(biomart3, "UKBID")

       UKBID  Count
0     P62805     14
1     P68431     10
2     P0C0S8      5
3     P23610      3
4     Q99666      2
...      ...    ...
3947  P22307      1
3948  Q16775      1
3949  Q14320      1
3950  Q9NP97      1
3951  Q9H204      1

[3952 rows x 2 columns]


In [84]:
biomart3[biomart3['UKBID'] == 'P62805']

Ensembl_Gene_description  Ensembl_Strand  \
148   histone cluster 1 H4 family member k [Source:H...              -1   
155   histone cluster 1 H4 family member j [Source:H...               1   
157   histone cluster 1 H4 family member l [Source:H...              -1   
3724  histone cluster 1 H4 family member a [Source:H...               1   
3726  histone cluster 1 H4 family member b [Source:H...              -1   
3792  histone cluster 1 H4 family member c [Source:H...               1   
4992  histone cluster 1 H4 family member i [Source:H...               1   
5359  histone cluster 1 H4 family member d [Source:H...              -1   
5381  histone cluster 1 H4 family member e [Source:H...               1   
5419  histone cluster 1 H4 family member f [Source:H...               1   
5442  histone cluster 1 H4 family member h [Source:H...              -1   
7252  histone cluster 2 H4 family member a [Source:H...               1   
7264  histone cluster 2 H4 family member b [Source:H...              -1   
8033  histone cluster 4 H4 [Source:HGNC Symbol;Acc:H...              -1   

       UKBID HGNC_symbol  
148   P62805    HIST1H4K  
155   P62805    HIST1H4J  
157   P62805    HIST1H4L  
3724  P62805    HIST1H4A  
3726  P62805    HIST1H4B  
3792  P62805    HIST1H4C  
4992  P62805    HIST1H4I  
5359  P62805    HIST1H4D  
5381  P62805    HIST1H4E  
5419  P62805    HIST1H4F  
5442  P62805    HIST1H4H  
7252  P62805    HIST2H4A  
7264  P62805    HIST2H4B  
8033  P62805     HIST4H4

In [89]:
# MERGE
MER = pd.merge(panther3953, biomart3, how='inner', on=['UKBID'])

MER

UKBID                                    GeneName_Symbol  \
0     O14737     Programmed cell death protein 5;PDCD5;ortholog   
1     P62195  26S proteasome regulatory subunit 8;PSMC5;orth...   
2     Q8IYB3  Serine/arginine repetitive matrix protein 1;SR...   
3     Q9UMY4                    Sorting nexin-12;SNX12;ortholog   
4     Q12849            G-rich sequence factor 1;GRSF1;ortholog   
...      ...                                                ...   
3986  Q15121                                                NaN   
3987  Q99470                                                NaN   
3988  Q9GZP8                                                NaN   
3989  Q9NRS6                                                NaN   
3990  Q9Y3B4                                                NaN   

                                  PANTHER_fam_subfamily  \
0       PROGRAMMED CELL DEATH PROTEIN 5 (PTHR10840:SF1)   
1     26S PROTEASOME REGULATORY SUBUNIT 8 (PTHR23073...   
2     SERINE/ARGININE REPETITIVE MATRIX PROTEIN 1 (P...   
3                      SORTING NEXIN-12 (PTHR45963:SF3)   
4             G-RICH SEQUENCE FACTOR 1 (PTHR13976:SF42)   
...                                                 ...   
3986                                                NaN   
3987                                                NaN   
3988                                                NaN   
3989                                                NaN   
3990                                                NaN   

                                     Molecular_function  \
0                                                   NaN   
1                 TBP-class protein binding(GO:0017025)   
2                              mRNA binding(GO:0003729)   
3                                                   NaN   
4     nucleic acid binding(GO:0003676);structural co...   
...                                                 ...   
3986                                                NaN   
3987                                                NaN   
3988                                                NaN   
3989                                                NaN   
3990                                                NaN   

                                     Biological_process  \
0                                                   NaN   
1     RNA polymerase II transcriptional preinitiatio...   
2                           mRNA processing(GO:0006397)   
3                                                   NaN   
4            mRNA splicing, via spliceosome(GO:0000398)   
...                                                 ...   
3986                                                NaN   
3987                                                NaN   
3988                                                NaN   
3989                                                NaN   
3990                                                NaN   

                                     Cellular_component  \
0               cytosol(GO:0005829);nucleus(GO:0005634)   
1     cytosolic part(GO:0044445);nuclear lumen(GO:00...   
2                                                   NaN   
3                                                   NaN   
4                                                   NaN   
...                                                 ...   
3986                                                NaN   
3987                                                NaN   
3988                                                NaN   
3989                                                NaN   
3990                                                NaN   

                        Protein_class  \
0           enzyme modulator(PC00095)   
1                  hydrolase(PC00121)   
2     mRNA processing factor(PC00147)   
3                                 NaN   
4          ribosomal protein(PC00202)   
...                               ...   
3986                              NaN   
3987                              NaN 

In [90]:
len(MER.UKBID.unique())

3952

In [91]:
MER.to_csv("MERGE_biomart_panther_proteinclass_3952Unique_ukbid.csv", index=False)